objects/o_player/Step_0.gml

```diff
@@ -30,6 +30,17 @@ if (x_speed_ > 0) {
 	image_xscale = -1;	
 }
 
+//reapply the fractions
+x_speed_ += x_speed_fraction;
+y_speed_ += y_speed_fraction;
```

Comment:
Putting "reapply the fractions" it here makes diagonal movement stutter.
Reapplying the fractions @ line 10 will cause a slowing on diagonal movement.